rmw_zenoh_cpp/src/detail/qos.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright 2024 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "qos.hpp"
+
+// Define defaults for various QoS settings.
+#define RMW_ZENOH_DEFAULT_HISTORY RMW_QOS_POLICY_HISTORY_KEEP_LAST;
+// If the depth field in the qos profile is set to 0, the RMW implementation
+// has the liberty to assign a default depth. The zenoh transport protocol
+// is configured with 256 channels so theoretically, this would be the maximum
+// depth we can set before blocking transport. A high depth would increase the
+// memory footprint of processes as more messages are stored in memory while a
+// very low depth might unintentionally drop messages leading to a poor
+// out-of-the-box experience for new users. For now we set the depth to 42,
+// a popular "magic number". See https://en.wikipedia.org/wiki/42_(number).
+#define RMW_ZENOH_DEFAULT_HISTORY_DEPTH 42;
+#define RMW_ZENOH_DEFAULT_RELIABILITY RMW_QOS_POLICY_RELIABILITY_RELIABLE;
+#define RMW_ZENOH_DEFAULT_DURABILITY RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL;
```

Comment:
This is a bit of a surprising default to me.  I would think that this would either be `RMW_QOS_POLICY_DURABILITY_SYSTEM_DEFAULT`, or, if we need this to be concrete, `RMW_QOS_POLICY_DURABILITY_VOLATILE`.  Can you explain the reasoning behind `TRANSIENT_LOCAL` here?

Reply:
The default is set in [rmf_dds_common::qos_profile_get_best_available_for_publisher/subscription](https://github.com/ros2/rmw_dds_common/blob/0cf1ce476af39f8706004ea5432d00645aa93a66/rmw_dds_common/src/qos.cpp#L500-L507). This function is used in `rmw_zenoh` prior to this PR and also in other DDS based RMWs.

Reply:
Ah, got it.  Thanks for the explanation, in that case, nothing to do here.